application/views/scripts/navigation/dashboard.phtml

```diff
@@ -1,11 +1,22 @@
+<?php
+
+use ipl\Html\HtmlString;
+
+?>
 <div class="controls">
     <?= $tabs ?>
 </div>
 <div class="content">
     <?php foreach ($navigation as $item): /** @var \Icinga\Web\Navigation\NavigationItem $item */?>
     <a class="dashboard-link" href="<?= $this->url($item->getUrl(), $item->getUrlParameters()) ?>"<?= $this->propertiesToString($item->getAttributes()) ?>>
         <div class="link-icon">
-            <?= $this->icon($item->getIcon() ?: 'forward', null, array('aria-hidden' => true)) ?>
+        <?php
+            if (gettype($item->getIcon()) == 'object' && get_class($item->getIcon()) == 'ipl\Web\Widget\Icon') {
```

Comment:
Please check the url to decide whether to use the old or new icons.

`if (substr($item->getUrl(), 0, 9) === 'icingadb/') {`

This way only Icinga DB Web should able to use them.

By the way, `$item->getIcon() instanceof \ipl\Web\Widget\Icon` would have been the correct check and loose comparisons (`==`) are discouraged most of the time, use strict comparsions (`===`) instead :wink:

Reply:
✅